client/ayon_core/plugins/publish/collect_anatomy_instance_data.py

```diff
@@ -411,13 +415,18 @@ def _fill_folder_data(self, instance, project_entity, anatomy_data):
             if hierarchy:
                 parent_name = hierarchy.split("/")[-1]
 
-            folder_name = instance.data["folderPath"].split("/")[-1]
+            folder_path = instance.data["folderPath"]
+            folder_name = folder_path.split("/")[-1]
             anatomy_data.update({
```

Comment:
If we implement `get_folder_template_data` in a way that it calls e.g. `get_folder_template_data_from_values(folder_path, folder_type, project_name)` then also here we could re-use the implementation of `ayon_core.template_data.get_folder_template_data_from_values` to ensure each of these end up using the same function instead of duplicating the code everywhere that seem to define these template data. Preferably there's one entry point to this. that returns the valid data - instead of having this duplicated.

Reply:
Can we focus more on the data keys?

Reply:
Might be better than to first fix the duplication of logic in a separate PR which would make the size of this PR change much smaller since there's much less to change.

Reply:
I created this PR to raise discussion about the keys....